test/addons/olm/start

```diff
@@ -21,21 +19,17 @@ def deploy(cluster):
     #   The CustomResourceDefinition "clusterserviceversions.operators.coreos.com"
     #   is invalid: metadata.annotations: Too long: must have at most 262144 bytes
     # See https://medium.com/pareture/kubectl-install-crd-failed-annotations-too-long-2ebc91b40c7d
-    kubectl.apply(
-        f"--filename={BASE_URL}/crds.yaml",
-        "--server-side=true",
-        context=cluster,
-    )
+    kubectl.apply("--kustomize", "crds", "--server-side=true", context=cluster)
 
     print("Waiting until cdrs are established")
     kubectl.wait(
         "--for=condition=established",
-        f"--filename={BASE_URL}/crds.yaml",
+        f"--filename=https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.22.0/crds.yaml",
         context=cluster,
     )
```

Comment:
What is the point of caching if we access the remote yaml again here?

I think we should try to simplify the installation by creating one kustomization like this one:
https://github.com/RamenDR/ramen/blob/f1fbf25623dd2da5a22d402a1aeba25f46ab0096/test/addons/csi-addons/kustomization.yaml

With both the crd and the operator yaml. With this we should be able to apply and cahce one kustomization and wait once.

I'm not sure this works, so we have to test it. Please remove the olm changes from this PR, since the rest of the changes are ready, and use another PR to work on olm. This is a good approach in general, not mix unrelated things in the same PR. Small pull requests are easier to review and can be merged quickly.